cloud/openssh/config_test.go

```diff
@@ -69,143 +69,227 @@ func TestHostOrMatchRegex(t *testing.T) {
 	}
 }
 
+//go:embed testdata/devbox-ssh-config.golden
+var goldenDevboxSSHConfig []byte
+
 func TestSetupDevbox(t *testing.T) {
-	testdirs := []string{
-		"testdata/no-config",
-		"testdata/existing-config",
-		"testdata/already-setup",
-	}
-	for _, testdata := range testdirs {
-		t.Run(testdata, func(t *testing.T) {
-			workdir := duplicateDir(t, filepath.Join(testdata, "in"))
-			t.Setenv("HOME", workdir)
-
-			if err := SetupDevbox(); err != nil {
-				t.Error("got SetupDevbox() error:", err)
-			}
-			dirsEqual(t, workdir, filepath.Join(testdata, "out"))
-		})
+	want := fstest.MapFS{
+		".config":            &fstest.MapFile{Mode: fs.ModeDir | 0755},
+		".config/devbox":     &fstest.MapFile{Mode: fs.ModeDir | 0755},
+		".config/devbox/ssh": &fstest.MapFile{Mode: fs.ModeDir | 0700},
+		".config/devbox/ssh/config": &fstest.MapFile{
+			Data: goldenDevboxSSHConfig,
+			Mode: 0644,
+		},
+		".config/devbox/ssh/known_hosts": &fstest.MapFile{
+			Data: sshKnownHosts,
+			Mode: 0644,
+		},
+
+		".ssh": &fstest.MapFile{Mode: fs.ModeDir | 0700},
+		".ssh/config": &fstest.MapFile{
+			Data: []byte("Include \"$HOME/.config/devbox/ssh/config\"\n"),
+			Mode: 0644,
+		},
 	}
+
+	t.Run("NoConfigs", func(t *testing.T) {
+		in := fstest.MapFS{}
+		workdir := fsToDir(t, in)
+		t.Setenv("HOME", workdir)
+		if err := SetupDevbox(); err != nil {
+			t.Error("got SetupDevbox() error:", err)
+		}
+		got := os.DirFS(workdir)
+		fsEqual(t, got, want)
+	})
+	t.Run("ExistingSSHConfig", func(t *testing.T) {
+		existingSSHConfig := []byte("Host example.com\n\tUser example\n\tPort 1234\n")
+		in := fstest.MapFS{
+			".ssh": &fstest.MapFile{Mode: fs.ModeDir | 0700},
+			".ssh/config": &fstest.MapFile{
+				Data: existingSSHConfig,
+				Mode: 0644,
+			},
+		}
+		// Temporarily change the desired ~/.ssh/config so it contains
+		// the initial contents of the input ~/.ssh/config.
+		originalWantConfig := want[".ssh/config"]
+		defer func() { want[".ssh/config"] = originalWantConfig }()
+		want[".ssh/config"] = &fstest.MapFile{
+			Data: append(
+				[]byte("Include \"$HOME/.config/devbox/ssh/config\"\n"),
+				existingSSHConfig...,
+			),
+			Mode: 0644,
+		}
```

Comment:
hmm does this affect whether these cases can run in parallel?

Assuming so:
- this is a non-concern for this testing scenario
- but if we move these utility functions to an `fstest` package then we may want to provide a facility for cloning `want` without modifying it directly so that test cases could run in parallel...

Reply:
It does, but they can't be run in parallel anyway because of `t.Setenv`. I think that the calling test would be responsible for setting up the wanted filesystem, so that probably wouldn't be a concern if we extracted the fs helpers.

Overall, I wasn't sure if it was more readable to swap out that single file or to duplicate `want` for each subtest. Let me know if you think copying `want` would make things clearer and I can update it.

Reply:
nah, this is fine for now. It is clear.